feat(vscode): support `oxlint --lsp`

Sysix opened the pull request:
Hacked a bit to test if the `node_modules/.bin/oxlint --lsp` will work:
```
{
  "oxc.path.server": "/home/sysix/dev/oxc/node_modules/.bin/oxlint"
}
```
<img width="887" height="196" alt="grafik" src="https://github.com/user-attachments/assets/76088dd1-3ede-4b83-86b9-09527edb784a" />

"works" with windows + shell:
<img width="1556" height="123" alt="grafik" src="https://github.com/user-attachments/assets/9753bf44-05cd-471f-b8a1-7c48d3436491" />

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15680?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#15680** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15680?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15680?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#15611** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15611?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15610** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15610?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15609** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15609?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15563** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15563?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 1 other dependent PR ([#15608](https://github.com/oxc-project/oxc/pull/15608) <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15608?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>)
* **#15505** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15505?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15568** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15568?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15680).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on editors/vscode/client/extension.ts:
```diff
@@ -166,6 +166,7 @@ export async function activate(context: ExtensionContext) {
         }
       : {
           command: path!,
```
can we remove the `process.env.OXLINT_LSP_TEST` now?

Sysix committed:
feat(vscode): support `oxlint --lsp` (#15680)

Hacked a bit to test if the `node_modules/.bin/oxlint --lsp` will work:
```
{
  "oxc.path.server": "/home/sysix/dev/oxc/node_modules/.bin/oxlint"
}
```
<img width="887" height="196" alt="grafik" src="https://github.com/user-attachments/assets/76088dd1-3ede-4b83-86b9-09527edb784a" />

"works" with windows + shell:
<img width="1556" height="123" alt="grafik" src="https://github.com/user-attachments/assets/9753bf44-05cd-471f-b8a1-7c48d3436491" />
